### ISSUE TITLE
DartPad doesn't support generic method syntax.

### ISSUE THREAD

**Issue by lrhn** [dart-lang/dart-pad#754]
According to dartPad itself, it is running Dart 1.21.0.
This should include the new generic method syntax, but neither the analyzer nor the dart2js compiler used by dartpad accepts generic method syntax.
See: https://dartpad.dartlang.org/587244c9cc9c86ec3551b13a4d4bed98

**Comment by lukechurch** [comment]
@lrhn Thanks for the report.

The analyzer isn't accepting the code as it currently needs a flag setting in analyzer options. I've opened a new issue here: https://github.com/dart-lang/dart-services/issues/280 I've made the change and am now runnning the extended test suite. I'm going to close this defect in favour of that one.

The compiler is working correctly, but looks like it isn't because the way the error is displayed is very unhelpful. If you open your link, click 'Run' and look in the console you'll see:

```
VM251:524 Uncaught Error: NoSuchMethodError: No top-level function named 'main'.
    at Object.wrapException (<anonymous>:524:17)
    at Object.missingMain (<anonymous>:537:15)
    at <anonymous>:831:9
    at init.currentScript (<anonymous>:815:7)
    at <anonymous>:826:5
    at <anonymous>:834:3
    at replaceJavaScript (VM46 frame.html:39)
    at VM46 frame.html:69
```

if you adjust to include a main method 

```
main() {
  new Z().bar();
}
```

You will see '5' as would be expected. The error UX is 'a trifle sub-optimal', but at least the compiler is behaving itself. I'll open a seperate issue for the error UX.
